bundles/net.tourbook/src/net/tourbook/data/TourData.java

```diff
@@ -8990,7 +8990,7 @@ public float[] getPowerSerie() {
 
       for (int timeIndex = 0; timeIndex < timeSerie.length; timeIndex++) {
 
-         final float speed = speedSerie[timeIndex] / 3.6f; // speed km/h -> m/s
+         final float speed = UI.convertSpeed_KmhToMs(speedSerie[timeIndex]);
```

Comment:
I did some testing with the [Flight Recorder](https://www.oracle.com/java/technologies/javase/products-jmc8-downloads.html) and imported [this file](https://github.com/mytourbook/mytourbook/blob/main/bundles/net.tourbook.tests/src/test/device/garmin/fit/files/ConeyLakeMove_2020_05_23_08_55_42_Trail%2Brunning.fit) and didn't find any bottle necks with or without `convertSpeed_KmhToMs`

![image](https://github.com/mytourbook/mytourbook/assets/25235665/bf4d0e98-5e41-41c5-bf1b-ff5200e60ab1)

>Very early in the MT development I discovered with the YourKit Java profiler,

How long ago was "very early" ? Now that we use Java 11, it's also possible that the JRE has improved.

I can revert if you really want me to but I personally don't see a big impact of performance and also I prefer having a unique function instead of duplicated code in several files

Reply:
> How long ago was "very early" ? 

About 15 years ago

I did a performance test with this result

```

   19878 Time slices

   speed = speedSerie[timeIndex] / 3.6f;

   0.1917 ms
   0.0413 ms
   0.0411 ms
   0.0769 ms
   0.0404 ms
   0.0420 ms
   0.0399 ms
   0.0415 ms
   0.0406 ms
   0.0404 ms
   0.0415 ms
   0.0439 ms
   0.0403 ms
   0.0405 ms
   0.0472 ms


   speed = UI.convertSpeed_KmhToMs(speedSerie[timeIndex]);     

   1.1513 ms
   0.1724 ms
   0.2007 ms
   0.0369 ms
   0.0688 ms
   0.0406 ms
   0.0366 ms
   0.0375 ms
   0.1298 ms
   0.0374 ms
   0.0385 ms
   0.0414 ms
   0.0362 ms
   0.0370 ms
   0.0375 ms

```

Test Environment:
- Java 17
- The tour cache is set to 2 that when reselecting 3 tours, then it is recomputed. Tour cache size 1 fails with an exception

So the initial 3 call of the same tour are much slower but then they are slightly faster. As this performance loss is not recognizable, we can keep the usage of `UI.convertSpeed_KmhToMs`

Reply:
> > How long ago was "very early" ?
> 
> About 15 years ago
> 

That's a long time for computers 😄 


> I did a performance test with this result
> 


That's a good test.
I think we also need to take into account that the compiler does optimizations behind the scenes that we don't see when looking at the raw code.